packages/examples/fortune-v2/job-launcher/server/src/modules/payment/payment.service.ts

```diff
@@ -0,0 +1,159 @@
+      email,
+    });
+
+    if (!customer) {
+      this.logger.log(ErrorPayment.CustomerNotFound, PaymentService.name);
+      throw new NotFoundException(ErrorPayment.CustomerNotFound);
+    }
+
+    return customer.id;
+  }
+
+  public async createFiatPayment(customerId: string, dto: PaymentFiatCreateDto) {
+    const { amount, currency } = dto;
+    const paymentMethodOptions = {};
+
+    const params: Stripe.PaymentIntentCreateParams = {
+      payment_method_types: [PaymentFiatMethodType.CARD],
+      amount: amount * 100,
+      currency: currency,
+    };
+
+    params.confirm = true;
+    params.customer = customerId;
+
+    if (paymentMethodOptions) {
+      params.payment_method_options = paymentMethodOptions;
+    }
+
+    const paymentIntent = await this.stripe.paymentIntents.create(params);
+
+    return {
+      clientSecret: paymentIntent.client_secret
+    };
+  }
+
+  public async confirmFiatPayment(userId: number, dto: PaymentFiatConfirmDto): Promise<boolean> {
+    try {
+      const paymentData = await this.getPayment(dto.paymentId);
+
+      if (paymentData?.status?.toUpperCase() !== PaymentStatus.SUCCEEDED) {
+        this.logger.log(ErrorPayment.NotSuccess, PaymentService.name);
+        throw new NotFoundException(ErrorPayment.NotSuccess);
+      }
+
+      await this.savePayment(userId, PaymentSource.FIAT, PaymentType.DEPOSIT, BigNumber.from(paymentData.amount))
+
+      return true;
+    } catch (e) {
+      this.logger.log(ErrorPayment.NotFound, PaymentService.name);
+      return false;
+    }
+  }
+
+  public async createCryptoPayment(userId: number, dto: PaymentCryptoCreateDto) {
+    try {
+      const signer = this.ethersSigner.createWallet(this.configService.get<string>("WEB3_JOB_LAUNCHER_PRIVATE_KEY", "web3 private key"));
+      
+      const transaction = await signer.provider.getTransactionReceipt(dto.transactionHash);
+      console.log(transaction)
+      if (!transaction) { 
+        this.logger.error(ErrorPayment.TransactionNotFoundByHash)
+        throw new NotFoundException(ErrorPayment.TransactionNotFoundByHash);
+      }
+
+      if (transaction.confirmations < TX_CONFIRMATION_TRESHOLD) {
+        this.logger.error(`Transaction has ${transaction.confirmations} confirmations instead of ${TX_CONFIRMATION_TRESHOLD}`)
+        throw new NotFoundException(ErrorPayment.TransactionHasNotEnoughAmountOfConfirmations);
+      } 
+
+      await this.savePayment(userId, PaymentSource.CRYPTO, PaymentType.DEPOSIT, BigNumber.from(dto.amount));
+      
+      return true;
+    } catch (e) {
+      this.logger.log(ErrorPayment.NotFound, PaymentService.name);
+      return false;
+    }
+  }
+
+  public async savePayment(userId: number, source: PaymentSource, type: PaymentType, amount: BigNumber): Promise<boolean> {
+    const rate = await this.currencyService.getRate(TokenId.HUMAN_PROTOCOL, Currency.USD)
+
+    const paymentEntity = await this.paymentRepository
```

Comment:
Pyament Entity needs to save the transaction hash, so we can validate that we add twice the same transaction